package.json

```
@@ -21,7 +23,36 @@
     "release": "npm run test && npm run build && npm run publish"
   },
   "devDependencies": {
+    "@types/jest": "^24.0.15",
+    "@types/node": "^10.14.7",
+    "@typescript-eslint/eslint-plugin": "^1.9.0",
+    "@typescript-eslint/parser": "^1.9.0",
+    "eslint": "^5.16.0",
+    "eslint-config-airbnb-base": "^13.1.0",
+    "eslint-config-prettier": "^6.0.0",
+    "eslint-loader": "^2.1.2",
+    "eslint-plugin-import": "^2.17.3",
+    "eslint-plugin-prettier": "^3.1.0",
+    "husky": "^3.0.1",
+    "prettier": "^1.17.1",
     "lerna": "^3.15.0",
-    "lerna-changelog": "^0.8.2"
+    "lerna-changelog": "^0.8.2",
+    "jest": "^24.8.0",
+    "ts-jest": "^24.0.2",
+    "lint-staged": "^9.2.0",
+    "rimraf": "^2.6.3",
+    "typescript": "^3.5.3"
+  },
+  "husky": {
+    "hooks": {
+      "pre-commit": "lint-staged"
+    }
+  },
+  "lint-staged": {
+    "**/*.{js, ts}": [
```

```suggestion
    "**/*.{js,ts}": [
````
另外怎么这里又不抽离 husky 和 lint-staged 配置项了？

---

显得 <root> 干净点。。。

---

靠... 一天一个变啊 大哥

---

拥抱变化。
改，提出去~

---

`.editorconfig` `.eslintignore` `.prettierignore` 等等这些配置文件只需要 root 里统一一份就行吧，像 `eslint` `prettier` 这些应该搞个组织共用的 config，这样基本只需要一行了，放在外面那么多 . 文件反而污染视线

---

可以的，就顶层放着其实也还行吧。。这些东西就是恶心人的

---

可以的，就顶层放着其实也还行吧。。这些东西就是恶心人的

---

> `.editorconfig` `.eslintignore` `.prettierignore` 等等这些配置文件只需要 root 里统一一份就行吧，像 `eslint` `prettier` 这些应该搞个组织共用的 config，这样基本只需要一行了，放在外面那么多 . 文件反而污染视线

这是个哲学问题。只能投票决定了。

---

> 这是个哲学问题。只能投票决定了。

哈哈

---

配置文件我是偏向提出去，好处是让人一眼看到项目有哪些配置的东西，坏处是乱一点。
配置的内容，我是偏向提供一个公用的 config, 另外一个 repo 在干这事。

---

@leohxj 就放顶层可以了，提出 repo 没必要